README.md

```diff
@@ -1,20 +1,221 @@
+
+## Spring YAML Housekeeping Configuration
+If your project utilises Spring YAML you can define your Housekeeping within the YAML. For example:
+
+    housekeeping:
+      data-source:
+        #The package of your driver class
+        driver-class-name: com.mysql.cj.jdbc.Driver
+        #JDBC URL for your Database
+        url: jdbc:mysql://housekeeping.foo1baz123.us-east-1.rds.amazonaws.com:3306/housekeeping_db
+        #Database Username
+        username: bdp
+        #Database Password
+        password: Ch4ll3ng3
+
+## Programatic Housekeeping Configuration
+Housekeeping allows you to configure your housekeeping job in a more fine grained manner by providing a certain set of Spring beans in your application.
+
+Users can configure their housekeeping data source in code by defining the bean `DataSource housekeepingDataSource(...)`.
+For example:
+
+      @Bean(destroyMethod = "close")
+      DataSource housekeepingDataSource(
+        String driverClassName,
+        String jdbcUrl,
+        String username,
+        String encryptedPassword) {
+        return DataSourceBuilder
+          .create()
+          .driverClassName(driverClassName)
+          .url(jdbcUrl)
+          .username(username)
+          .password(encryptedPassword)
+          .build();
+      }
+
+Housekeeping comes with a default `HousekeepingService` implementation, however the user can choose to provide their own implementation. To
+run housekeeping you must provide a `HousekeepingService` bean which either constructs the default `FileSystemHousekeepingService` or a custom implementation of the `HousekeepingService` interface.
+
+      @Bean
+      HousekeepingService housekeepingService(
+          LegacyReplicaPathRepository legacyReplicaPathRepository) {
+        return new FileSystemHousekeepingService(legacyReplicaPathRepository, new org.apache.hadoop.conf.Configuration());
+      }
+
+The housekeeping support provided defaults to creating a database of the name `housekeeping` and a table of the name `legacy_replica_path` within which housekeeping data is stored. To enable this database the user must provide a schema.sql file on their classpath. This file contains any SQL code which must be run to initialise your database upon application startup. This is particularly important if running Housekeeping in your application for the first time.
+
+An example schema.sql file for use with the default housekeeping entity configuration is given below:
+
+      CREATE SCHEMA IF NOT EXISTS housekeeping;
+
+
+Applications which leverage housekeeping support can define their own schema and table which they desire their Housekeeping data to be stored into. This can be acheived in three steps.
+
+The user must create their database initialisation schema.sql script, and add it to their classpath (or alternatively provide it as a resource in your application).
```

Comment:
In other places you refer to the first person "you" so stick with this instead of the third person "the user", it's confusing if you switch between the two.